test cases/common/277 custom target private dir/meson.build

```diff
@@ -0,0 +1,14 @@
+project('277 custom target private dir')
+
+python = find_program('python3', required: true)
+args = [
+  '-c',
+  'import os, sys; os.chdir(sys.argv[1])',
+]
+
+custom_target(
+  'check-private-dir',
+  command: [python, args, '@PRIVATE_DIR@'],
+  output: 'check-private-dir',
```

Comment:
Might have your little script actually create this file just for completeness

Reply:
Sounds good, updated the test case with feedback above along with a print and `capture: true` so this file actually gets created.

Reply:
I think it should create the file directly instead of using `capture: true` since that kwarg will cause the command line for the custom target to be saved into a second (python) script, which meson internally generates to handle the capturing part.

It means the test case runs two python programs instead of one, and contributes a bit more to the time that it takes to complete the test run on systems with slow forking, such as windows 

Admittedly the overhead of adding an entire new test case directory is larger than forking an extra python copy, and so is the overhead of detecting MSVC (which this case doesn't do but many other unittests do a little too frequently). Still, I'm not sure there's a good solution at the moment for those.

Reply:
The test case truly does not need to the create the output file to test the functionality of this fix. I feel like if I add more code to this to write out that file (instead of just using `capture`, which itself isn't even necessary for the test) it would detract from focusing on the actual reduction of what this is meant to test.

If you disagree though, please let me know and I'll be happy to update with writing to `@OUTPUT@` to hopefully get this merged.